src/annotations/AnnotationFile.php

```diff
@@ -60,6 +60,12 @@ class AnnotationFile
     public $uses;
 
     /**
+     * @var string[] $losingMethods hash mapping fully-qualified class name to
+     * the list of fully-qualifed methods that lost insteadof inclusion.
+     */
+    public $losingMethods;
```

Comment:
I think it would be better to store all mapping from parsing `insteadof` and `as` in this array and call it `traitMethodOverrides` instead. Then we at runtime:

* include all trait methods, that aren't listed in that array
* include scan array and manually add method overrides/aliases from corresponding traits